shapeshift/web
# fix: redirect to an error modal when the recovery seed phrase is wrong

## Description
- Added a `RecoveryInvalid` modal opened when error code === 3 (SYNTAXERROR)
- On click on the `Retry Again` button, it execute the `recover` method from `hdwallet` again, meaning that the user is getting back to the `Pin` modal. It looks like we're pretty forced to do this, AFAIK we can't get back to the beginning of the seed entries directly


## Notice




## Pull Request Type


## Issue (if applicable)
closes #1618


## Risk
- Related to the keepkey recover feature, it means that introducing a bug there could block users from recovering a keepkey wakket


## Testing
- Wipe a keepkey (be sure that you have the recover seed phrase somewhere)
- Connect the keepkey and enter the recover workflow
- Type any wrong word to the end
- See the error modal, click on `Retry Again`
- Enter your pin again, enter the right seed phrase
- It should be successfull :+1: 


## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/14963751/166108195-3c018166-7186-49de-9ce0-0089e86bbfb7.png)
